whoops you have no retail addons

**Describe the bug**
Tukui.org is currently having issues im guessing from the pre-patch user traffic.  The site is not down but the connection appears to timeout.  I suspect that ajour is trying to reach out to check if there is an update for elvui and is either erroring or timing out.  It seems to either timeout and display addons or fails and gives a whoops cant load addons message. Also the time before this timeout is really long (around 2-3 minutes) probably because aujor is waiting for a response from tukui and doesnt have a timeout?  If it does load addons elvui is not in the list which confirms its not loading the site.

EDIT: error in the log confirms it.

**To Reproduce**
Steps to reproduce the behavior:
1. have elvui installed
2. block connection to tukui.org (this might not work?)
3. refresh your addons


**Expected behavior**
A timeout after a few seconds of being unable to connect to tukui.org

**Screenshots**
![image](https://user-images.githubusercontent.com/72519738/95926268-b3dc7580-0d70-11eb-927f-c0b9e9833637.png)


**Software involved**
Please complete the following information:
 - OS: Windows 10
 - Ajour version: 0.4.1
 - (if applicable) Addon name: any flavor of elvui

**Additional context**
Add any other context about the problem here.

**Log Output**
16:10:39.318 [ajour][INFO] Ajour 0.4.1 has started.
16:10:39.320 [ajour::gui][DEBUG] antialiasing: true
16:10:39.339 [ajour_core::config][DEBUG] loading config
16:10:39.339 [ajour_core::utility][DEBUG] checking for application update
16:10:39.339 [ajour_core::theme][DEBUG] loading user themes
16:10:39.340 [ajour_core::fs::theme][DEBUG] loaded 0 user themes
16:10:39.968 [ajour::gui::update][DEBUG] Message::Parse
16:10:39.968 [ajour::gui::update][DEBUG] config loaded:
Config {
    wow: Wow {
        directory: Some(
            "G:\\World of Warcraft",
        ),
        flavor: Retail,
    },
    addons: Addons {
        ignored: {
            Retail: [],
            Classic: [],
        },
        release_channels: {
            Retail: {},
            Classic: {},
        },
    },
    theme: None,
    column_config: V1 {
        local_version_width: 150,
        remote_version_width: 150,
        status_width: 85,
    },
    window_size: Some(
        (
            900,
            620,
        ),
    ),
    scale: None,
    backup_directory: Some(
        "G:\\WoW Addon Backup",
    ),
}
16:10:39.969 [ajour::gui::update][DEBUG] preparing to parse addons in "G:\\World of Warcraft\\_retail_\\Interface/AddOns"
16:10:39.969 [ajour::gui::update][DEBUG] preparing to parse addons in "G:\\World of Warcraft\\_classic_\\Interface/AddOns"
16:10:39.969 [ajour::gui::update][DEBUG] Message::ThemesLoaded(0 themes)
16:10:39.969 [ajour::gui::update][DEBUG] Message::NeedsUpdate(None)
16:10:39.969 [ajour_core::parse][DEBUG] retail - parsing addons folder
16:10:39.969 [ajour_core::parse][DEBUG] classic - parsing addons folder
16:10:39.969 [ajour_core::parse][DEBUG] retail - 136 folders in AddOns directory to parse
16:10:39.969 [ajour_core::parse][DEBUG] classic - 63 folders in AddOns directory to parse
16:10:39.977 [ajour::gui::update][DEBUG] Message::LatestBackup(Some(2020-10-13T16:03:40))
16:10:40.133 [ajour::gui::update][DEBUG] Message::CatalogDownloaded(8595 addons in catalog)
16:10:41.675 [ajour_core::parse][DEBUG] classic - 63 fingerprints: 62 cached, 1 calculated, 1 added, 0 removed
16:10:41.697 [ajour_core::parse][DEBUG] classic - 63 successfully parsed from '.toc'
16:10:41.697 [ajour_core::parse][DEBUG] classic - 2 addons with tukui id
16:10:45.659 [ajour::gui::update][DEBUG] Interaction::Settings
16:10:48.844 [ajour::gui::update][DEBUG] Message::ThemeSelected("Outrun")
16:10:49.531 [ajour_core::parse][DEBUG] retail - 136 fingerprints: 136 cached, 0 calculated, 0 added, 0 removed
16:10:49.562 [ajour_core::parse][DEBUG] retail - 134 successfully parsed from '.toc'
16:10:49.562 [ajour_core::parse][DEBUG] retail - 4 addons with tukui id
16:10:52.185 [ajour::gui::update][DEBUG] Message::ThemeSelected("Sort")
16:10:56.735 [ajour::gui::update][DEBUG] Message::ThemeSelected("Dracula")
16:10:58.846 [ajour::gui::update][DEBUG] Message::ThemeSelected("Ayu")
16:11:00.458 [ajour::gui::update][DEBUG] Message::ThemeSelected("Gruvbox")
16:11:02.043 [ajour::gui::update][DEBUG] Message::ThemeSelected("Nord")
16:11:03.567 [ajour::gui::update][DEBUG] Message::ThemeSelected("Solarized Dark")
16:11:05.385 [ajour::gui::update][DEBUG] Message::ThemeSelected("Solarized Light")
16:11:07.546 [ajour::gui::update][DEBUG] Message::ThemeSelected("Horde")
16:11:09.895 [ajour::gui::update][DEBUG] Message::ThemeSelected("Alliance")
16:11:11.325 [ajour::gui::update][DEBUG] Message::ThemeSelected("Horde")
16:11:13.548 [ajour::gui::update][DEBUG] Interaction::Settings
16:11:41.702 [ajour_core::parse][DEBUG] classic - 0 addons applied with tukui id package metadata
16:11:41.702 [ajour_core::parse][DEBUG] classic - 61 unique fingerprints to check against curse api
16:12:08.470 [ajour_core::parse][DEBUG] classic - 23 exact fingerprint matches against curse api
16:12:08.471 [ajour_core::parse][DEBUG] classic - 23 addons applied with fingerprint metadata
16:12:08.471 [ajour_core::parse][DEBUG] classic - 23 addons with curse id
16:12:38.495 [ajour_core::parse][DEBUG] classic - 23 addons successfully parsed
16:12:38.495 [ajour::gui::update][DEBUG] Message::ParsedAddons(classic, 23 addons)
16:12:38.553 [ajour_core::parse][DEBUG] retail - 1 addons applied with tukui id package metadata
16:12:38.553 [ajour_core::parse][DEBUG] retail - 130 unique fingerprints to check against curse api
16:12:43.064 [ajour::gui::update][ERROR] Message::ParsedAddons(retail) - invalid type: null, expected a sequence at line 1 column 68007
16:12:47.361 [ajour::gui::update][DEBUG] Interaction::Refresh
16:12:47.361 [ajour_core::config][DEBUG] loading config
16:12:47.363 [ajour::gui::update][DEBUG] Message::Parse
16:12:47.363 [ajour::gui::update][DEBUG] config loaded:
Config {
    wow: Wow {
        directory: Some(
            "G:\\World of Warcraft",
        ),
        flavor: Retail,
    },
    addons: Addons {
        ignored: {
            Classic: [],
            Retail: [],
        },
        release_channels: {
            Classic: {},
            Retail: {},
        },
    },
    theme: Some(
        "Horde",
    ),
    column_config: V1 {
        local_version_width: 150,
        remote_version_width: 150,
        status_width: 85,
    },
    window_size: Some(
        (
            900,
            620,
        ),
    ),
    scale: None,
    backup_directory: Some(
        "G:\\WoW Addon Backup",
    ),
}
16:12:47.363 [ajour::gui::update][DEBUG] preparing to parse addons in "G:\\World of Warcraft\\_retail_\\Interface/AddOns"
16:12:47.363 [ajour::gui::update][DEBUG] preparing to parse addons in "G:\\World of Warcraft\\_classic_\\Interface/AddOns"
16:12:47.363 [ajour_core::parse][DEBUG] retail - parsing addons folder
16:12:47.363 [ajour_core::parse][DEBUG] classic - parsing addons folder
16:12:47.363 [ajour_core::parse][DEBUG] classic - 63 folders in AddOns directory to parse
16:12:47.363 [ajour_core::parse][DEBUG] retail - 136 folders in AddOns directory to parse
16:12:47.363 [ajour::gui::update][DEBUG] Message::LatestBackup(Some(2020-10-13T16:03:40))
16:12:47.369 [ajour_core::parse][DEBUG] retail - 136 fingerprints: 136 cached, 0 calculated, 0 added, 0 removed
16:12:47.397 [ajour_core::parse][DEBUG] retail - 134 successfully parsed from '.toc'
16:12:47.397 [ajour_core::parse][DEBUG] retail - 4 addons with tukui id
16:12:47.402 [ajour_core::parse][DEBUG] classic - 63 fingerprints: 63 cached, 0 calculated, 0 added, 0 removed
16:12:47.427 [ajour_core::parse][DEBUG] classic - 63 successfully parsed from '.toc'
16:12:47.427 [ajour_core::parse][DEBUG] classic - 2 addons with tukui id
16:13:25.519 [ajour_core::parse][DEBUG] classic - 1 addons applied with tukui id package metadata
16:13:25.519 [ajour_core::parse][DEBUG] classic - 61 unique fingerprints to check against curse api
16:13:51.416 [ajour_core::parse][DEBUG] classic - 23 exact fingerprint matches against curse api
16:13:51.417 [ajour_core::parse][DEBUG] classic - 23 addons applied with fingerprint metadata
16:13:51.417 [ajour_core::parse][DEBUG] classic - 23 addons with curse id
16:14:21.422 [ajour_core::parse][DEBUG] classic - 24 addons successfully parsed
16:14:21.423 [ajour::gui::update][DEBUG] Message::ParsedAddons(classic, 24 addons)
16:14:27.698 [ajour_core::parse][DEBUG] retail - 0 addons applied with tukui id package metadata
16:14:27.698 [ajour_core::parse][DEBUG] retail - 130 unique fingerprints to check against curse api
16:14:57.720 [ajour::gui::update][ERROR] Message::ParsedAddons(retail) - Couldn't fetch details for addon. Server returned: <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">
<TITLE>ERROR: The request could not be satisfied</TITLE>
</HEAD><BODY>
<H1>504 ERROR</H1>
<H2>The request could not be satisfied.</H2>
<HR noshade size="1px">
CloudFront attempted to establish a connection with the origin, but either the attempt failed or the origin closed the connection.
We can't connect to the server for this app or website at this time. There might be too much traffic or a configuration error. Try again later, or contact the app or website owner.
<BR clear="all">
If you provide content to customers through CloudFront, you can find steps to troubleshoot and help prevent this error by reviewing the CloudFront documentation.
<BR clear="all">
<HR noshade size="1px">
<PRE>
Generated by cloudfront (CloudFront)
Request ID: 1edcKN2iOtXXEauutrxWPlLQidNF7L8_a9BJfExND4MstigFCGYowQ==
</PRE>
<ADDRESS>
</ADDRESS>
</BODY></HTML>
16:15:26.108 [ajour::gui::update][DEBUG] Interaction::FlavorSelected(retail)
16:15:29.034 [ajour::gui::update][DEBUG] Interaction::Settings
16:15:31.987 [ajour::gui::update][DEBUG] Message::ThemeSelected("Outrun")
16:15:34.061 [ajour::gui::update][DEBUG] Interaction::Settings
16:15:34.938 [ajour::gui::update][DEBUG] Interaction::Settings
16:15:36.811 [ajour::gui::update][DEBUG] Message::ThemeSelected("Sort")
16:15:38.438 [ajour::gui::update][DEBUG] Interaction::Settings
16:15:42.927 [ajour::gui::update][DEBUG] Interaction::Refresh
16:15:42.928 [ajour_core::config][DEBUG] loading config
16:15:42.929 [ajour::gui::update][DEBUG] Message::Parse
16:15:42.929 [ajour::gui::update][DEBUG] config loaded:
Config {
    wow: Wow {
        directory: Some(
            "G:\\World of Warcraft",
        ),
        flavor: Retail,
    },
    addons: Addons {
        ignored: {
            Retail: [],
            Classic: [],
        },
        release_channels: {
            Classic: {},
            Retail: {},
        },
    },
    theme: Some(
        "Sort",
    ),
    column_config: V1 {
        local_version_width: 150,
        remote_version_width: 150,
        status_width: 85,
    },
    window_size: Some(
        (
            900,
            620,
        ),
    ),
    scale: None,
    backup_directory: Some(
        "G:\\WoW Addon Backup",
    ),
}
16:15:42.929 [ajour::gui::update][DEBUG] preparing to parse addons in "G:\\World of Warcraft\\_retail_\\Interface/AddOns"
16:15:42.929 [ajour::gui::update][DEBUG] preparing to parse addons in "G:\\World of Warcraft\\_classic_\\Interface/AddOns"
16:15:42.930 [ajour_core::parse][DEBUG] retail - parsing addons folder
16:15:42.930 [ajour_core::parse][DEBUG] classic - parsing addons folder
16:15:42.930 [ajour_core::parse][DEBUG] classic - 63 folders in AddOns directory to parse
16:15:42.930 [ajour_core::parse][DEBUG] retail - 136 folders in AddOns directory to parse
16:15:42.930 [ajour::gui::update][DEBUG] Message::LatestBackup(Some(2020-10-13T16:03:40))
16:15:42.933 [ajour_core::parse][DEBUG] classic - 63 fingerprints: 63 cached, 0 calculated, 0 added, 0 removed
16:15:42.948 [ajour_core::parse][DEBUG] classic - 63 successfully parsed from '.toc'
16:15:42.948 [ajour_core::parse][DEBUG] classic - 2 addons with tukui id
16:15:42.952 [ajour_core::parse][DEBUG] retail - 136 fingerprints: 136 cached, 0 calculated, 0 added, 0 removed
16:15:42.977 [ajour_core::parse][DEBUG] retail - 134 successfully parsed from '.toc'
16:15:42.977 [ajour_core::parse][DEBUG] retail - 4 addons with tukui id
16:16:42.952 [ajour_core::parse][DEBUG] classic - 0 addons applied with tukui id package metadata
16:16:42.952 [ajour_core::parse][DEBUG] classic - 61 unique fingerprints to check against curse api
16:16:59.630 [ajour_core::parse][DEBUG] retail - 1 addons applied with tukui id package metadata
16:16:59.630 [ajour_core::parse][DEBUG] retail - 130 unique fingerprints to check against curse api
16:17:01.666 [ajour_core::parse][TRACE] retail - partial fingerprint found:
	Curse ID: 3054074
	File Name: DBM-Core-8.3.33-69-g37b889a
	Modules (Name - Local Fingerprint - Remote Fingerprint)
	DBM-Core - 3007858050 - 3631944795
	DBM-Brawlers - 856194762 - 856194762
	DBM-DMF - 82871402 - 82871402
	DBM-GUI - 4082440440 - 4082440440
	DBM-WorldEvents - 175674435 - 175674435
	DBM-Party-BfA - 1720199172 - 1720199172
	DBM-Party-Shadowlands - 4145805302 - 4145805302
	DBM-Azeroth-BfA - 1437758332 - 1437758332
	DBM-Uldir - 1121508959 - 1121508959
	DBM-CrucibleofStorms - 2732257158 - 2732257158
	DBM-ZuldazarRaid - 2261956449 - 2261956449
	DBM-EternalPalace - 2307249171 - 2307249171
	DBM-Nyalotha - 665230461 - 665230461
	DBM-CastleNathria - 1297925907 - 1297925907
	DBM-DefaultSkin - 460709238 - 460709238
	DBM-StatusBarTimers - 28633784 - 28633784
16:17:01.666 [ajour_core::parse][TRACE] retail - partial fingerprint found:
	Curse ID: 3079016
	File Name: Kui_Nameplates-beta-flying-pig-mechanical-pony-5-gd6c30e4
	Modules (Name - Local Fingerprint - Remote Fingerprint)
	Kui_Nameplates_Core - 3424327693 - 1167733820
	Kui_Nameplates_Core_Config - 3399312238 - 3526822007
	Kui_Nameplates - 3160421675 - 2792397089
	Kui_Media - 2125922415 - 2125922415
16:17:01.666 [ajour_core::parse][TRACE] retail - partial fingerprint found:
	Curse ID: 3018307
	File Name: Kui_Nameplates-beta-talking-horse-sprinting-whale
	Modules (Name - Local Fingerprint - Remote Fingerprint)
	Kui_Nameplates_Core - 3424327693 - 2776017104
	Kui_Nameplates_Core_Config - 3399312238 - 52168374
	Kui_Nameplates - 3160421675 - 3160421675
	Kui_Media - 2125922415 - 2125922415
16:17:01.666 [ajour_core::parse][TRACE] retail - partial fingerprint found:
	Curse ID: 2999417
	File Name: MythicDungeonTools-v3.0.96-alpha
	Modules (Name - Local Fingerprint - Remote Fingerprint)
	MethodDungeonTools - 744117555 - 744117555
	MythicDungeonTools - 1849219519 - 4240916830
16:17:01.666 [ajour_core::parse][TRACE] retail - partial fingerprint found:
	Curse ID: 2869636
	File Name: OmniCC 8.3.4
	Modules (Name - Local Fingerprint - Remote Fingerprint)
	OmniCC_Config - 1625150679 - 1625150679
	OmniCC - 3042965591 - 949952694
16:17:01.666 [ajour_core::parse][TRACE] retail - partial fingerprint found:
	Curse ID: 3074634
	File Name: RaiderIO-v202010080600
	Modules (Name - Local Fingerprint - Remote Fingerprint)
	RaiderIO_DB_EU_A - 4150387447 - 4150387447
	RaiderIO_DB_EU_A_R - 2134664830 - 2134664830
	RaiderIO_DB_EU_H - 3476472942 - 3476472942
	RaiderIO_DB_EU_H_R - 1757014448 - 1757014448
	RaiderIO_DB_KR_A - 3217259566 - 3217259566
	RaiderIO_DB_KR_A_R - 1972602044 - 1972602044
	RaiderIO_DB_KR_H - 318390728 - 318390728
	RaiderIO_DB_KR_H_R - 2806785340 - 2806785340
	RaiderIO_DB_TW_A - 2626748412 - 2626748412
	RaiderIO_DB_TW_A_R - 1452455250 - 1452455250
	RaiderIO_DB_TW_H - 2351492670 - 2351492670
	RaiderIO_DB_TW_H_R - 3168521648 - 3168521648
	RaiderIO_DB_US_A - 1489670592 - 1489670592
	RaiderIO_DB_US_A_R - 2438231150 - 2438231150
	RaiderIO_DB_US_H - 100953872 - 100953872
	RaiderIO_DB_US_H_R - 4248814126 - 4248814126
	RaiderIO - 3649062825 - 1165398449
16:17:01.666 [ajour_core::parse][DEBUG] retail - 24 exact fingerprint matches against curse api
16:17:01.667 [ajour_core::parse][DEBUG] retail - 24 addons applied with fingerprint metadata
16:17:01.667 [ajour_core::parse][DEBUG] retail - 24 addons with curse id
16:17:04.972 [ajour_core::parse][DEBUG] retail - 24 addons applied with curse id package metadata
16:17:04.972 [ajour_core::parse][DEBUG] retail - 25 addons successfully parsed
16:17:04.972 [ajour::gui::update][DEBUG] Message::ParsedAddons(retail, 25 addons)
16:17:13.087 [ajour::gui::update][ERROR] Message::ParsedAddons(classic) - Couldn't fetch details for addon. Server returned: <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">
<TITLE>ERROR: The request could not be satisfied</TITLE>
</HEAD><BODY>
<H1>504 ERROR</H1>
<H2>The request could not be satisfied.</H2>
<HR noshade size="1px">
CloudFront attempted to establish a connection with the origin, but either the attempt failed or the origin closed the connection.
We can't connect to the server for this app or website at this time. There might be too much traffic or a configuration error. Try again later, or contact the app or website owner.
<BR clear="all">
If you provide content to customers through CloudFront, you can find steps to troubleshoot and help prevent this error by reviewing the CloudFront documentation.
<BR clear="all">
<HR noshade size="1px">
<PRE>
Generated by cloudfront (CloudFront)
Request ID: r8_BZpdSlP60iO-JsdeQyAh1YhQWPLVCjcuAyqglTbf3bwDx1cF_-w==
</PRE>
<ADDRESS>
</ADDRESS>
</BODY></HTML>
16:18:35.607 [ajour::gui::update][DEBUG] Interaction::UpdateAll
16:18:35.608 [ajour_core::network][DEBUG] downloading remote version Release-r1241 for Ace3
16:18:35.608 [ajour_core::network][DEBUG] downloading remote version 1.0.4 for AceGUI-3.0_SFX-Widgets
16:18:35.608 [ajour_core::network][DEBUG] downloading remote version 3.09.41.Retail for ArkInventory
16:18:35.608 [ajour_core::network][DEBUG] downloading remote version 8.3.1 for Masque
16:18:35.608 [ajour_core::network][DEBUG] downloading remote version 9.0.0 for Masque_Caith
16:18:35.608 [ajour_core::network][DEBUG] downloading remote version 1.20.1 for NameplateSCT
16:18:35.608 [ajour_core::network][DEBUG] downloading remote version v9.0.1-01 for Simulationcraft
16:18:35.608 [ajour_core::network][DEBUG] downloading remote version 3.0.0 for WeakAuras
16:18:35.609 [ajour_core::network][DEBUG] downloading remote version World Quests List v.9.3 for WorldQuestsList
16:18:36.073 [ajour::gui::update][DEBUG] Message::DownloadedAddon((retail, Masque_Caith))
16:18:36.075 [ajour::gui::update][DEBUG] Message::DownloadedAddon((retail, AceGUI-3.0_SFX-Widgets))
16:18:36.097 [ajour::gui::update][DEBUG] Message::DownloadedAddon((retail, NameplateSCT))
16:18:36.125 [ajour::gui::update][DEBUG] Message::DownloadedAddon((retail, Simulationcraft))
16:18:36.125 [ajour::gui::update][DEBUG] Message::DownloadedAddon((retail, Ace3))
16:18:36.157 [ajour::gui::update][DEBUG] Message::UnpackedAddon((AceGUI-3.0_SFX-Widgets, error: false))
16:18:36.158 [ajour_core::parse][DEBUG] retail - updating fingerprint for AceGUI-3.0_SFX-Widgets
16:18:36.161 [ajour::gui::update][DEBUG] Message::DownloadedAddon((retail, Masque))
16:18:36.192 [ajour::gui::update][DEBUG] Message::UpdateFingerprint((Retail, AceGUI-3.0_SFX-Widgets, error: false))
16:18:36.251 [ajour::gui::update][DEBUG] Message::UnpackedAddon((Masque_Caith, error: false))
16:18:36.253 [ajour_core::parse][DEBUG] retail - updating fingerprint for Masque_Caith
16:18:36.268 [ajour::gui::update][DEBUG] Message::UpdateFingerprint((Retail, Masque_Caith, error: false))
16:18:36.357 [ajour::gui::update][DEBUG] Message::DownloadedAddon((retail, WorldQuestsList))
16:18:36.456 [ajour::gui::update][DEBUG] Message::UnpackedAddon((Simulationcraft, error: false))
16:18:36.457 [ajour_core::parse][DEBUG] retail - updating fingerprint for Simulationcraft
16:18:36.481 [ajour::gui::update][DEBUG] Message::DownloadedAddon((retail, ArkInventory))
16:18:36.492 [ajour::gui::update][DEBUG] Message::UpdateFingerprint((Retail, Simulationcraft, error: false))
16:18:36.524 [ajour::gui::update][DEBUG] Message::UnpackedAddon((WorldQuestsList, error: false))
16:18:36.525 [ajour_core::parse][DEBUG] retail - updating fingerprint for WorldQuestsList
16:18:36.545 [ajour::gui::update][DEBUG] Message::UpdateFingerprint((Retail, WorldQuestsList, error: false))
16:18:36.606 [ajour::gui::update][DEBUG] Message::UnpackedAddon((NameplateSCT, error: false))
16:18:36.607 [ajour_core::parse][DEBUG] retail - updating fingerprint for NameplateSCT
16:18:36.635 [ajour::gui::update][DEBUG] Message::UnpackedAddon((Ace3, error: false))
16:18:36.636 [ajour_core::parse][DEBUG] retail - updating fingerprint for Ace3
16:18:36.663 [ajour::gui::update][DEBUG] Message::UpdateFingerprint((Retail, NameplateSCT, error: false))
16:18:36.693 [ajour::gui::update][DEBUG] Message::UpdateFingerprint((Retail, Ace3, error: false))
16:18:36.952 [ajour::gui::update][DEBUG] Message::UnpackedAddon((Masque, error: false))
16:18:36.953 [ajour_core::parse][DEBUG] retail - updating fingerprint for Masque
16:18:37.024 [ajour::gui::update][DEBUG] Message::UpdateFingerprint((Retail, Masque, error: false))
16:18:37.709 [ajour::gui::update][DEBUG] Message::UnpackedAddon((ArkInventory, error: false))
16:18:37.710 [ajour_core::parse][DEBUG] retail - updating fingerprint for ArkInventoryConfig
16:18:37.710 [ajour_core::parse][DEBUG] retail - updating fingerprint for ArkInventorySearch
16:18:37.710 [ajour_core::parse][DEBUG] retail - updating fingerprint for ArkInventoryRules_Example
16:18:37.710 [ajour_core::parse][DEBUG] retail - updating fingerprint for ArkInventoryRules
16:18:37.710 [ajour_core::parse][DEBUG] retail - updating fingerprint for ArkInventory
16:18:37.726 [ajour::gui::update][DEBUG] Message::UpdateFingerprint((Retail, ArkInventoryRules_Example, error: false))
16:18:37.734 [ajour::gui::update][DEBUG] Message::UpdateFingerprint((Retail, ArkInventoryConfig, error: false))
16:18:37.742 [ajour::gui::update][DEBUG] Message::UpdateFingerprint((Retail, ArkInventorySearch, error: false))
16:18:37.748 [ajour::gui::update][DEBUG] Message::UpdateFingerprint((Retail, ArkInventoryRules, error: false))
16:18:37.786 [ajour::gui::update][DEBUG] Message::DownloadedAddon((retail, WeakAuras))
16:18:37.854 [ajour::gui::update][DEBUG] Message::UpdateFingerprint((Retail, ArkInventory, error: false))
16:18:40.550 [ajour::gui::update][DEBUG] Message::UnpackedAddon((WeakAuras, error: false))
16:18:40.551 [ajour_core::parse][DEBUG] retail - updating fingerprint for WeakAuras
16:18:40.551 [ajour_core::parse][DEBUG] retail - updating fingerprint for WeakAurasArchive
16:18:40.551 [ajour_core::parse][DEBUG] retail - updating fingerprint for WeakAurasModelPaths
16:18:40.551 [ajour_core::parse][DEBUG] retail - updating fingerprint for WeakAurasTemplates
16:18:40.551 [ajour_core::parse][DEBUG] retail - updating fingerprint for WeakAurasOptions
16:18:40.565 [ajour::gui::update][DEBUG] Message::UpdateFingerprint((Retail, WeakAurasArchive, error: false))
16:18:40.576 [ajour::gui::update][DEBUG] Message::UpdateFingerprint((Retail, WeakAurasTemplates, error: false))
16:18:40.591 [ajour::gui::update][DEBUG] Message::UpdateFingerprint((Retail, WeakAurasModelPaths, error: false))
16:18:40.609 [ajour::gui::update][DEBUG] Message::UpdateFingerprint((Retail, WeakAurasOptions, error: false))
16:18:40.690 [ajour::gui::update][DEBUG] Message::UpdateFingerprint((Retail, WeakAuras, error: false))
16:19:06.370 [ajour::gui::update][DEBUG] Interaction::Refresh
16:19:06.370 [ajour_core::config][DEBUG] loading config
16:19:06.371 [ajour::gui::update][DEBUG] Message::Parse
16:19:06.372 [ajour::gui::update][DEBUG] config loaded:
Config {
    wow: Wow {
        directory: Some(
            "G:\\World of Warcraft",
        ),
        flavor: Retail,
    },
    addons: Addons {
        ignored: {
            Retail: [],
            Classic: [],
        },
        release_channels: {
            Classic: {},
            Retail: {},
        },
    },
    theme: Some(
        "Sort",
    ),
    column_config: V1 {
        local_version_width: 150,
        remote_version_width: 150,
        status_width: 85,
    },
    window_size: Some(
        (
            900,
            620,
        ),
    ),
    scale: None,
    backup_directory: Some(
        "G:\\WoW Addon Backup",
    ),
}
16:19:06.372 [ajour::gui::update][DEBUG] preparing to parse addons in "G:\\World of Warcraft\\_retail_\\Interface/AddOns"
16:19:06.372 [ajour::gui::update][DEBUG] preparing to parse addons in "G:\\World of Warcraft\\_classic_\\Interface/AddOns"
16:19:06.372 [ajour_core::parse][DEBUG] retail - parsing addons folder
16:19:06.372 [ajour_core::parse][DEBUG] classic - parsing addons folder
16:19:06.372 [ajour_core::parse][DEBUG] classic - 63 folders in AddOns directory to parse
16:19:06.372 [ajour_core::parse][DEBUG] retail - 136 folders in AddOns directory to parse
16:19:06.373 [ajour::gui::update][DEBUG] Message::LatestBackup(Some(2020-10-13T16:03:40))
16:19:06.375 [ajour_core::parse][DEBUG] classic - 63 fingerprints: 63 cached, 0 calculated, 0 added, 0 removed
16:19:06.392 [ajour_core::parse][DEBUG] classic - 63 successfully parsed from '.toc'
16:19:06.392 [ajour_core::parse][DEBUG] classic - 2 addons with tukui id
16:19:06.398 [ajour_core::parse][DEBUG] retail - 136 fingerprints: 136 cached, 0 calculated, 0 added, 0 removed
16:19:06.425 [ajour_core::parse][DEBUG] retail - 134 successfully parsed from '.toc'
16:19:06.425 [ajour_core::parse][DEBUG] retail - 4 addons with tukui id
16:19:48.457 [ajour_core::parse][DEBUG] classic - 0 addons applied with tukui id package metadata
16:19:48.457 [ajour_core::parse][DEBUG] classic - 61 unique fingerprints to check against curse api
16:19:50.719 [ajour_core::parse][DEBUG] classic - 23 exact fingerprint matches against curse api
16:19:50.719 [ajour_core::parse][DEBUG] classic - 23 addons applied with fingerprint metadata
16:19:50.719 [ajour_core::parse][DEBUG] classic - 23 addons with curse id
16:20:02.445 [ajour_core::parse][DEBUG] classic - 22 addons applied with curse id package metadata
16:20:02.445 [ajour_core::parse][DEBUG] classic - 23 addons successfully parsed
16:20:02.446 [ajour::gui::update][DEBUG] Message::ParsedAddons(classic, 23 addons)
16:20:57.460 [ajour_core::parse][DEBUG] retail - 0 addons applied with tukui id package metadata
16:20:57.460 [ajour_core::parse][DEBUG] retail - 130 unique fingerprints to check against curse api
16:21:27.606 [ajour::gui::update][ERROR] Message::ParsedAddons(retail) - Couldn't fetch details for addon. Server returned: <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">
<TITLE>ERROR: The request could not be satisfied</TITLE>
</HEAD><BODY>
<H1>504 ERROR</H1>
<H2>The request could not be satisfied.</H2>
<HR noshade size="1px">
CloudFront attempted to establish a connection with the origin, but either the attempt failed or the origin closed the connection.
We can't connect to the server for this app or website at this time. There might be too much traffic or a configuration error. Try again later, or contact the app or website owner.
<BR clear="all">
If you provide content to customers through CloudFront, you can find steps to troubleshoot and help prevent this error by reviewing the CloudFront documentation.
<BR clear="all">
<HR noshade size="1px">
<PRE>
Generated by cloudfront (CloudFront)
Request ID: rfQ4DstmAkLFxFZlO7h_JZlIzrrYSk9bH1C2MaLG4t77PDBdWMql2w==
</PRE>
<ADDRESS>
</ADDRESS>
</BODY></HTML>

I am having the same issue.
OS: Ubuntu Linux 20.04
Ajour version: 0.4.1

```
20:54:18.295 [ajour_core::parse][DEBUG] classic - parsing addons folder
20:54:18.295 [ajour_core::parse][DEBUG] retail - parsing addons folder
20:54:18.295 [ajour_core::parse][DEBUG] classic - 43 folders in AddOns directory to parse
20:54:18.295 [ajour_core::parse][DEBUG] retail - 70 folders in AddOns directory to parse
20:54:18.313 [ajour_core::parse][DEBUG] retail - 70 fingerprints: 70 cached, 0 calculated, 0 added, 0 removed
20:54:18.602 [ajour_core::parse][DEBUG] retail - 70 successfully parsed from '.toc'
20:54:18.602 [ajour_core::parse][DEBUG] retail - 0 addons with tukui id
20:54:18.602 [ajour_core::parse][DEBUG] retail - 0 addons applied with tukui id package metadata
20:54:18.602 [ajour_core::parse][DEBUG] retail - 70 unique fingerprints to check against curse api
20:54:18.602 [ajour_core::parse][DEBUG] classic - 43 fingerprints: 43 cached, 0 calculated, 0 added, 0 removed
20:54:18.801 [ajour_core::parse][DEBUG] classic - 43 successfully parsed from '.toc'
20:54:18.801 [ajour_core::parse][DEBUG] classic - 0 addons with tukui id
20:54:18.801 [ajour_core::parse][DEBUG] classic - 0 addons applied with tukui id package metadata
20:54:18.801 [ajour_core::parse][DEBUG] classic - 43 unique fingerprints to check against curse api
20:54:20.484 [ajour::gui::update][ERROR] Message::ParsedAddons(classic) - invalid type: null, expected a sequence at line 1 column 32946
20:54:21.115 [ajour::gui::update][ERROR] Message::ParsedAddons(retail) - invalid type: null, expected a sequence at line 1 column 52773
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 The sections should follow the order `Packaging`, `Added`, `Changed`, `Fixed` and `Removed`.
 
 ## [Unreleased]
+## [0.4.3] - 2020-10-22
+
+### Fixed
+- Fixed the CurseForge API issue.
+- Minor improvements to the matching addons which was caused by yesterdays update (SORRY!).
+
 ## [0.4.2] - 2020-10-21
 ### Added
 - Add fallback measures for displaying addons when fingerprinting fails or we can't link an addon to a remote registry.

--- a/crates/core/src/addon.rs
+++ b/crates/core/src/addon.rs
@@ -74,6 +74,9 @@ pub enum AddonState {
     Downloading,
     Fingerprint,
     Unpacking,
+    // TODO: I have currently removed the state where curse-id only addons become corrupt.
+    // It can happen that the fingerprint is unknown to the API but everything else is good.
+    // This is properly not the best solution going forward, but for now it solves the purpose.
     Corrupted,
     Updatable,
 }
@@ -314,9 +317,7 @@ impl Addon {
     }
 
     /// Creates an `Addon` from the Curse package. This is a fallback for when we don't
-    /// have an exact fingerprint match, but we have a curse id for the addon. Since we
-    /// can't guarantee which local version the addon is, we will set this addon status
-    /// as Corrupted and the local version as "Unknown".
+    /// have an exact fingerprint match, but we have a curse id for the addon.
     pub fn from_curse_package(
         package: &curse_api::Package,
         flavor: Flavor,
@@ -413,7 +414,6 @@ impl Addon {
             .cloned()
             .collect();
         addon.folders = folders;
-        addon.state = AddonState::Corrupted;
 
         Some(addon)
     }

--- a/crates/core/src/addon.rs
+++ b/crates/core/src/addon.rs
@@ -321,7 +321,7 @@ impl Addon {
         package: &curse_api::Package,
         flavor: Flavor,
         addon_folders: &[AddonFolder],
-    ) -> Self {
+    ) -> Option<Self> {
         let mut remote_packages = HashMap::new();
 
         let mut stable_exists = false;
@@ -371,7 +371,7 @@ impl Addon {
         } else if alpha_exists {
             3
         } else {
-            unreachable!("No file in curse package for {}", package.id);
+            return None;
         };
 
         let file = package
@@ -415,7 +415,7 @@ impl Addon {
         addon.folders = folders;
         addon.state = AddonState::Corrupted;
 
-        addon
+        Some(addon)
     }
 
     /// Creates an `Addon` from the Curse fingerprint info

--- a/crates/core/src/curse_api.rs
+++ b/crates/core/src/curse_api.rs
@@ -10,6 +10,7 @@ use isahc::prelude::*;
 use serde::{Deserialize, Serialize};
 
 const API_ENDPOINT: &str = "https://addons-ecs.forgesvc.net/api/v2";
+const FINGERPRINT_API_ENDPOINT: &str = "https://hub.dev.wowup.io/curseforge/addons/fingerprint";
 
 #[derive(Clone, Debug, Deserialize)]
 #[serde(rename_all = "camelCase")]
@@ -94,9 +95,21 @@ pub struct AddonFingerprintInfo {
     pub latest_files: Vec<File>,
 }
 
+#[derive(Serialize)]
+struct FingerprintData {
+    fingerprints: Vec<u32>,
+}
+
 pub async fn fetch_remote_packages_by_fingerprint(fingerprints: &[u32]) -> Result<FingerprintInfo> {
-    let url = format!("{}/fingerprint", API_ENDPOINT);
-    let mut resp = post_json_async(url, fingerprints, vec![], None).await?;
+    let mut resp = post_json_async(
+        FINGERPRINT_API_ENDPOINT,
+        FingerprintData {
+            fingerprints: fingerprints.to_owned(),
+        },
+        vec![],
+        None,
+    )
+    .await?;
     if resp.status().is_success() {
         let fingerprint_info = resp.json()?;
         Ok(fingerprint_info)

--- a/crates/core/src/curse_api.rs
+++ b/crates/core/src/curse_api.rs
@@ -33,7 +33,6 @@ pub struct File {
     pub game_version_flavor: Option<String>,
     pub modules: Vec<Module>,
     pub is_alternate: bool,
-    pub game_version_date_released: String,
     pub game_version: Vec<String>,
 }
 
@@ -83,13 +82,8 @@ pub struct CategorySection {
 #[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct FingerprintInfo {
-    pub is_cache_built: bool,
     pub exact_matches: Vec<AddonFingerprintInfo>,
-    pub exact_fingerprints: Vec<u32>,
     pub partial_matches: Vec<AddonFingerprintInfo>,
-    pub partial_match_fingerprints: ::serde_json::Value,
-    pub installed_fingerprints: Vec<u32>,
-    pub unmatched_fingerprints: Vec<u32>,
 }
 
 #[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]

--- a/crates/core/src/parse.rs
+++ b/crates/core/src/parse.rs
@@ -375,10 +375,6 @@ pub async fn read_addon_directory<P: AsRef<Path>>(
         .map(|info| Addon::from_curse_fingerprint_info(info.id, &info, flavor, &addon_folders))
         .collect();
 
-    for fa in &fingerprint_addons {
-        addon_folders.retain(|af| af.id != fa.primary_folder_id);
-    }
-
     log::debug!(
         "{} - {} addons from fingerprint metadata",
         flavor,
@@ -396,6 +392,11 @@ pub async fn read_addon_directory<P: AsRef<Path>>(
     // using the curse id from the `.toc`
     let mut curse_ids_from_nonmatch: Vec<_> = addon_folders
         .iter()
+        .filter(|f| {
+            fingerprint_addons
+                .iter()
+                .any(|fa| fa.primary_folder_id != f.id)
+        })
         .filter(|f| {
             f.repository_identifiers.tukui.is_none() && f.repository_identifiers.curse.is_some()
         })
@@ -457,9 +458,10 @@ pub async fn read_addon_directory<P: AsRef<Path>>(
                 || curse_ids_from_partial.contains(&package.id)
             {
                 let addon = Addon::from_curse_package(&package, flavor, &addon_folders);
-                curse_id_only_addons.push(addon);
-
-                created += 1;
+                if let Some(addon) = addon {
+                    curse_id_only_addons.push(addon);
+                    created += 1;
+                }
             }
         }
 

--- a/crates/core/src/parse.rs
+++ b/crates/core/src/parse.rs
@@ -480,9 +480,9 @@ pub async fn read_addon_directory<P: AsRef<Path>>(
 
     // Concats the different repo addons, and returns.
     let mut concatenated = [
+        &tukui_addons[..],
         &fingerprint_addons[..],
         &curse_id_only_addons[..],
-        &tukui_addons[..],
     ]
     .concat();
 
@@ -516,6 +516,21 @@ pub async fn read_addon_directory<P: AsRef<Path>>(
     log::debug!("{} - {} unknown addons", flavor, unknown_addons.len());
 
     concatenated.extend(unknown_addons);
+
+    // We do a extra clean up here to ensure that we dont display any addons which is a module.
+    // Idea is that we loop the addon.folders and mark all "dependencies".
+    // We then retain them from the concatenated Vec. If there is any.
+    let mut marked = vec![];
+    for addon in &concatenated {
+        for folder in &addon.folders {
+            if folder.id != addon.primary_folder_id {
+                marked.push(folder.id.clone());
+            }
+        }
+    }
+
+    // Remove dependency addons.
+    concatenated.retain(|addon| !marked.iter().any(|id| &addon.primary_folder_id == id));
 
     Ok(concatenated)
 }

--- a/crates/core/src/parse.rs
+++ b/crates/core/src/parse.rs
@@ -375,6 +375,10 @@ pub async fn read_addon_directory<P: AsRef<Path>>(
         .map(|info| Addon::from_curse_fingerprint_info(info.id, &info, flavor, &addon_folders))
         .collect();
 
+    for fa in &fingerprint_addons {
+        addon_folders.retain(|af| af.id != fa.primary_folder_id);
+    }
+
     log::debug!(
         "{} - {} addons from fingerprint metadata",
         flavor,

